### PR TITLE
custom config supported

### PR DESCRIPTION
**Pull request by hendrikebbers**
- custom config supported
- guava removed

---
This change is [Reviewable](https://reviewable.io/reviews/canoo/dolphin-platform/852)

### CODE REVIEW

File: platform/dolphin-platform-client/src/main/java/com/canoo/dp/impl/platform/client/config/ConfigurationFileLoader.java [hunk 1]
```diff
@@ -49,28 +51,55 @@ public static DefaultClientConfiguration loadConfiguration() {
...
         return configuration;
     }
 
-    private static DefaultClientConfiguration createConfiguration() {
+    private static DefaultClientConfiguration createConfiguration(final String... additionalLocations) {
         try {
             final ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
 
+            if(additionalLocations != null) {
+                for(String location : additionalLocations) {
```
Comment (giftkugel) [inline platform/dolphin-platform-client/src/main/java/com/canoo/dp/impl/platform/client/config/ConfigurationFileLoader.java]: Add `final` keyword.

File: platform/dolphin-platform-client/src/main/java/com/canoo/dp/impl/platform/client/config/ConfigurationFileLoader.java [hunk 2]
```diff
@@ -49,28 +51,55 @@ public static DefaultClientConfiguration loadConfiguration() {
...
+                    LOG.trace("Trying to load configuration at '" + location + "'");
+                    try (final InputStream inputStream = classLoader.getResourceAsStream(location)) {
+                        if (inputStream != null) {
+                            return readConfig(inputStream);
+                        }
+                    } catch (Exception e) {
```
Comment (giftkugel) [inline platform/dolphin-platform-client/src/main/java/com/canoo/dp/impl/platform/client/config/ConfigurationFileLoader.java]: Add `final` keyword.

File: platform/dolphin-platform-client/src/main/java/com/canoo/dp/impl/platform/client/config/ConfigurationFileLoader.java [hunk 3]
```diff
@@ -49,28 +51,55 @@ public static DefaultClientConfiguration loadConfiguration() {
...
+                    }
+                }
+            }
+
+            try (final InputStream inputStream = classLoader.getResourceAsStream(PLATFORM_LOCATION)) {
+                LOG.trace("Trying to load configuration at '" + PLATFORM_LOCATION + "'");
+                if (inputStream != null) {
+                    return readConfig(inputStream);
+                }
+            } catch (Exception e) {
```
Comment (giftkugel) [inline platform/dolphin-platform-client/src/main/java/com/canoo/dp/impl/platform/client/config/ConfigurationFileLoader.java]: Add `final` keyword.

File: platform/dolphin-platform-client/src/main/java/com/canoo/dp/impl/platform/client/config/ConfigurationFileLoader.java [hunk 4]
```diff
@@ -49,28 +51,55 @@ public static DefaultClientConfiguration loadConfiguration() {
...
+                LOG.trace("No config found at '" + PLATFORM_LOCATION + "'");
+            }
+
             try (final InputStream inputStream = classLoader.getResourceAsStream(DEFAULT_LOCATION)) {
+                LOG.trace("Trying to load configuration at '" + DEFAULT_LOCATION + "'");
                 if (inputStream != null) {
                     return readConfig(inputStream);
                 }
+            } catch (Exception e) {
```
Comment (giftkugel) [inline platform/dolphin-platform-client/src/main/java/com/canoo/dp/impl/platform/client/config/ConfigurationFileLoader.java]: Add `final` keyword.

File: platform/dolphin-platform-client/src/main/java/com/canoo/dp/impl/platform/client/config/ConfigurationFileLoader.java [hunk 5]
```diff
@@ -49,28 +51,55 @@ public static DefaultClientConfiguration loadConfiguration() {
...
+                LOG.trace("No config found at '" + DEFAULT_LOCATION + "'");
             }
 
             try (final InputStream inputStream = classLoader.getResourceAsStream(JAR_LOCATION)) {
+                LOG.trace("Trying to load configuration at '" + JAR_LOCATION + "'");
                 if (inputStream != null) {
                     return readConfig(inputStream);
                 }
+            } catch (Exception e) {
```
Comment (giftkugel) [inline platform/dolphin-platform-client/src/main/java/com/canoo/dp/impl/platform/client/config/ConfigurationFileLoader.java]: Add `final` keyword.

File: platform/dolphin-platform-client/src/main/java/com/canoo/dp/impl/platform/client/config/ConfigurationFileLoader.java [hunk 6]
```diff
@@ -49,28 +51,55 @@ public static DefaultClientConfiguration loadConfiguration() {
...
+                LOG.trace("No config found at '" + JAR_LOCATION + "'");
             }
 
             return new DefaultClientConfiguration();
-        } catch (IOException e) {
+        } catch (Exception e) {
```
Comment (giftkugel) [inline platform/dolphin-platform-client/src/main/java/com/canoo/dp/impl/platform/client/config/ConfigurationFileLoader.java]: Add `final` keyword.

File: platform/dolphin-platform-client/src/main/java/com/canoo/dp/impl/platform/client/config/ConfigurationFileLoader.java (hunk 1 above)
Comment (hendrikebbers) [reply platform/dolphin-platform-client/src/main/java/com/canoo/dp/impl/platform/client/config/ConfigurationFileLoader.java]: done

File: platform/dolphin-platform-client/src/main/java/com/canoo/dp/impl/platform/client/config/ConfigurationFileLoader.java (hunk 2 above)
Comment (hendrikebbers) [reply platform/dolphin-platform-client/src/main/java/com/canoo/dp/impl/platform/client/config/ConfigurationFileLoader.java]: done

File: platform/dolphin-platform-client/src/main/java/com/canoo/dp/impl/platform/client/config/ConfigurationFileLoader.java (hunk 3 above)
Comment (hendrikebbers) [reply platform/dolphin-platform-client/src/main/java/com/canoo/dp/impl/platform/client/config/ConfigurationFileLoader.java]: blöd

File: platform/dolphin-platform-client/src/main/java/com/canoo/dp/impl/platform/client/config/ConfigurationFileLoader.java (hunk 4 above)
Comment (hendrikebbers) [reply platform/dolphin-platform-client/src/main/java/com/canoo/dp/impl/platform/client/config/ConfigurationFileLoader.java]: done

File: platform/dolphin-platform-client/src/main/java/com/canoo/dp/impl/platform/client/config/ConfigurationFileLoader.java (hunk 5 above)
Comment (hendrikebbers) [reply platform/dolphin-platform-client/src/main/java/com/canoo/dp/impl/platform/client/config/ConfigurationFileLoader.java]: done

File: platform/dolphin-platform-client/src/main/java/com/canoo/dp/impl/platform/client/config/ConfigurationFileLoader.java (hunk 6 above)
Comment (hendrikebbers) [reply platform/dolphin-platform-client/src/main/java/com/canoo/dp/impl/platform/client/config/ConfigurationFileLoader.java]: done

**Pull request merged**